gengokit/middlewares/middlewares_test.go

```diff
@@ -0,0 +1,236 @@
+package middlewares
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/TuneLab/go-truss/gengokit"
+	thelper "github.com/TuneLab/go-truss/gengokit/gentesthelper"
+	"github.com/TuneLab/go-truss/svcdef"
+)
+
+var gopath []string
+
+func init() {
+	gopath = filepath.SplitList(os.Getenv("GOPATH"))
+}
+
+func TestNewServiceMiddleware(t *testing.T) {
+	var wantService = `
+		package middlewares
+
+		import (
+		pb "github.com/TuneLab/go-truss/gengokit/general-service"
+		)
+
+		func InjectServiceMiddlewares(in pb.ProtoServiceServer) pb.ProtoServiceServer {
+			return in
+		}
+	`
+
+	_, data, err := generalService()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	middleware := New()
+	service, err := middleware.Render(ServicePath, data)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	serviceBytes, err := ioutil.ReadAll(service)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wantFormatted, serviceFormatted, diff := thelper.DiffGoCode(wantService, string(serviceBytes))
+	if wantFormatted != serviceFormatted {
+		t.Fatalf("Serivce middleware different than expected:\n\n%s", diff)
+	}
+}
+
+func TestNewEndpoitnMiddleware(t *testing.T) {
+	var wantEndpoints = `
+		package middlewares
+
+		import (
+			"github.com/go-kit/kit/endpoint"
+		)
+
+		func InjectEndpointMiddlewares(in endpoint.Endpoint) endpoint.Endpoint {
+			return in
+		}
+	`
+
+	_, data, err := generalService()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	middleware := New()
+	endpoints, err := middleware.Render(EndpointsPath, data)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	endpointsBytes, err := ioutil.ReadAll(endpoints)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wantFormatted, endpointFormatted, diff := thelper.DiffGoCode(wantEndpoints, string(endpointsBytes))
+	if wantFormatted != endpointFormatted {
+		t.Fatalf("Endpoints middleware different than expected:\n\n%s", diff)
+	}
+}
+
+func TestRenderPrevService(t *testing.T) {
+	var wantService = `
+		package middlewares
+
+		import (
+		pb "github.com/TuneLab/go-truss/gengokit/general-service"
+		)
+
+		func InjectServiceMiddlewares(in pb.ProtoServiceServer) pb.ProtoServiceServer {
+			return in
+		}
+
+		func FooBar() error {
+			return nil
+		}
+	`
+	_, data, err := generalService()
+	if err != nil {
+		t.Fail()
+	}
+
+	middleware := New()
+
+	middleware.LoadService(strings.NewReader(wantService))
+
+	service, err := middleware.Render(ServicePath, data)
+	if err != nil {
+		t.Fail()
```

Comment:
Is `Fail` actually what you want here? Mark the test as failed but continue execution? Seems like an error like this would prevent any further execution of the test from being worthwhile.